java/api-test/integration-tests/src/test/java/com/microsoft/semantickernel/e2e/KernelTest.java

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft. All rights reserved.
+package com.microsoft.semantickernel.e2e;
+
+import com.microsoft.openai.OpenAIAsyncClient;
+import com.microsoft.semantickernel.Kernel;
+import com.microsoft.semantickernel.KernelConfig;
+import com.microsoft.semantickernel.builders.SKBuilders;
+import com.microsoft.semantickernel.connectors.ai.openai.textcompletion.OpenAITextCompletion;
+import com.microsoft.semantickernel.textcompletion.CompletionSKContext;
+import com.microsoft.semantickernel.textcompletion.CompletionSKFunction;
+import com.microsoft.semantickernel.textcompletion.TextCompletion;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledIf;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class KernelTest extends AbstractKernelTest {
+    private static final Logger LOGGER = LoggerFactory.getLogger(KernelTest.class);
+
+    @Test
+    @EnabledIf("isAzureTestEnabled")
+    public void endToEndTextGenerationTestAzureOpenAI() throws IOException {
+        Kernel kernel = buildKernel(getAzureOpenAIAPI(), "text-davinci-003");
+        executeChatCompletion(kernel);
+    }
+
+    @Test
+    @EnabledIf("isAzureTestEnabled")
+    public void endToEndTextGenerationTestOpenAI() throws IOException {
+        Kernel kernel = buildKernel(getOpenAIAPI(), getOpenAiModel());
+        executeChatCompletion(kernel);
```

Comment:
The kernel was built using text completion so `executeChatCompletion` looks wrong here.